src/themes/teams/components/Avatar/avatarStyles.ts

```diff
@@ -49,7 +50,7 @@ const getAvatarFontSize = (size: number) => {
 }
 
 const avatarStyles: IComponentPartStylesInput = {
-  root: ({ props: { size } }): ICSSInJSStyle => ({
+  root: ({ props: { size } }: { props: IAvatarProps }): ICSSInJSStyle => ({
```

Comment:
seems that it is better for us to introduce general type for that as well, instead of inline type literal. Let me take a closer look to propose a suggestion

Reply:
Well generally the the object that we are receiving here is {props, variables} so we may introduce this, but the thing is this is not used everywhere, in some cases we have just the props, or nothing which isn't a problem, but when we have just the pros or nothing we immediately know that this part of the styles is somewhat static and doesn't depend on any prop of variable. So I am not sure if we wan to specify specific types which may won't be used...

Reply:
Should we not wait until we figure out the optimal typing pattern for styles and variables before typing them?  Especially as we plan to change this signature.

We haven't yet figured out how 3rd parties will create themes, but we know they will need access to component typings as well so that they can type their styles/variables as well.  Not sure what the best way to type these but I know from testing it myself that there are several ways to go about it, each with their own pros and cons.

I would propose that PRs opened against #117 for the purpose of typing props do not also attempt to type themes, just props for now.

Reply:
lets try to follow the thought to see if we are on the same page. Seems that, at least semantically, the type of functions that are used for styling is the following (names are introduced just as an example):

```js
type StyleProvider<TProps, TVariables = any> = (args: StyleProviderArg) => ICSSInJSStyle`

type StyleProviderArg<TProps, TVariables = any> = { props: TProps, variables: TVariables }
```

To me it seems reasonable to declare this knowledge explicitly - because otherwise we would have the following issues for the following form:

```js
root: ({ props: { size } }: { props: IAvatarProps }): ICSSInJSStyle =>
```

- it is unclear for developer that variables parameter can be also consumed
- an attempt to consume variables parameter will result in immediate need to change literal type. This will quickly become annoying with each new case :)

```js
// compile-time error! type literal needs to be fixed as well
root: ({ props: { size }, variables }: { props: IAvatarProps }): ICSSInJSStyle =>
```

- also an important factor that it will be much easier to read the following expression
```js
// compile-time error! type literal needs to be fixed as well
root: ({ props: { size }}: StyleProviderArg<IAvatarProps>): ICSSInJSStyle =>
```

Please, let me know what do you think

Reply:
spoke with Levi, he has suggested to address this one in a set of separate PRs.

Reply:
agreed to resolve this problem later, by means of dedicated PRs set